server/game/cards/06-CotE/CourtlyChallenger.js

```diff
@@ -0,0 +1,51 @@
+const DrawCard = require('../../drawcard.js');
+const { Players, CardTypes } = require('../../Constants');
+
+class CourtlyChallenger extends DrawCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            effect: [
+                ability.effects.delayedEffect({
+                    when: {
+                        afterDuel: (event) => event.winner && event.winner === this
+                    },
+                    message: '{0} is honored due to winning a duel',
+                    gameAction: ability.actions.honor(context => ({ target: context.source }))
+                })
+            ]
+        });
+        this.persistentEffect({
```

Comment:
Is there a specific reason for separating the two persistent effects? I'd prefer them to be together, just in case there is ever an ability which copies a specific constant ability